src/status_im2/common/contacts/events.cljs

```diff
@@ -0,0 +1,100 @@
+(ns status-im2.common.contacts.events
+  (:require [utils.re-frame :as rf]
+            [status-im.utils.types :as types]
+            [re-frame.core :as re-frame]
+            [status-im.ethereum.core :as ethereum]
+            [status-im.ethereum.ens :as ens]
+            [status-im.ethereum.stateofus :as stateofus]
+            [status-im.native-module.core :as status]
+            [status-im.utils.db :as utils.db]
+            [status-im.utils.utils :as utils]))
+
+(re-frame/reg-fx
+ :resolve-public-key-from-ens-name
+ (fn [{:keys [chain-id ens-name on-success on-error]}]
+   (ens/pubkey chain-id ens-name on-success on-error)))
+
+(re-frame/reg-fx
+ :decompress-public-key
+ (fn [{:keys [public-key on-success on-error]}]
+   (status/compressed-key->public-key
+    public-key
+    (fn [resp]
+      (let [{:keys [error]} (types/json->clj resp)]
+        (if error ; If response is of the form {:error "foo"} => ERROR
+          (on-error error)
+          (on-success ; ...and replace "fe701" prefix with "0x" prefix
+           (str "0x" (subs resp 5)))))))))
+
+(rf/defn set-new-identity
+  {:events [:new-contact/set-new-identity]}
+  [{:keys [db]} input]
+  (let [input           (utils/safe-trim input)
+        public-key?     (utils.db/valid-public-key? input)
+        compressed-key? (utils.db/valid-compressed-key? input)
+        ens-name        (if compressed-key? nil (stateofus/ens-name-parse input))
+        on-success      (fn [pubkey]
+                          (rf/dispatch
+                           [:new-contact/set-new-identity-success
+                            input ens-name pubkey]))
+        on-error        (fn [err]
+                          (rf/dispatch
+                           [:new-contact/set-new-identity-error err input]))]
+    (cond
+      ;; clear identity from db
```

Comment:
I think these comments are all explaining *what* is being done, but the code speaks for itself.

Reply:
done